products/compute/api.yaml

```diff
@@ -856,11 +838,13 @@ objects:
         description: |
           Indicates whether the backend service will be used with internal or
```

Comment:
If I understand this correctly, the GCP API could allow you to create regional load-balancers through the BackendService API, but we're making an active choice to force people to use the RegionalBackendService API (which overlaps with the regular BackendService API)

Reply:
Yeah, I've traced that back while trying to understand that change in version 2.4.0 of the terraform provider:

- https://github.com/hashicorp/terraform/issues/14178#issuecomment-299266389
- https://github.com/hashicorp/terraform/pull/14301/files
- https://github.com/danawillow/terraform/commit/3b710368773b3ca79bb93dd40826c195016cea59#diff-831a88c8efb982906373493c41d0911f
- https://github.com/hashicorp/terraform/pull/10903

Reply:
@rambleraptor, that doesn't quite match my understanding. Pulling my message from Slack earlier today:

> The same base `BackendService` object is used in both the global and region APIs; so while the documentation indicates the field is usable, I don’t believe that’s actually true of this API.

To my understanding, the `region` field has never done anything on the global BackendService endpoint, and it's only read in the regional endpoint. That's because `region` is a __parameter__ in the URL and not a property in the request body.

(On the regional endpoint, it's true that it's a **response** property, and part of the Google API spec is that read-only properties can be sent to the API and they are just ignored instead of causing errors, in case users do a Read, modify the body, then do an Update/Patch. This also lead to the field being ignored in all cases on the global resource, because the definition is shared.)

Initially, as in @pdecat's findings, we'd incorrectly included the `region` field on the global resource because of the shared object definition and then removed it. That said, if you can find a counterexample I am very open to my understanding being proven wrong here.

Reply:
Cool, that seems more in-line with what I would expect from the API!